exporter/dynatraceexporter/internal/serialization/summary.go

```diff
@@ -0,0 +1,66 @@
+package serialization
+
+import (
+	dtMetric "github.com/dynatrace-oss/dynatrace-metric-utils-go/metric"
+	"github.com/dynatrace-oss/dynatrace-metric-utils-go/metric/dimensions"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+func serializeSummaryPoint(name, prefix string, dims dimensions.NormalizedDimensionList, dp pmetric.SummaryDataPoint) (string, error) {
+	if dp.Count() == 0 {
+		return "", nil
+	}
+
+	min, max, sum := summaryDataPointToSummary(dp)
+
+	dm, err := dtMetric.NewMetric(
+		name,
+		dtMetric.WithPrefix(prefix),
+		dtMetric.WithDimensions(dims),
+		dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+		dtMetric.WithFloatSummaryValue(min, max, sum, int64(dp.Count())),
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return dm.Serialize()
+}
+
+func serializeSummary(logger *zap.Logger, prefix string, metric pmetric.Metric, defaultDimensions dimensions.NormalizedDimensionList, staticDimensions dimensions.NormalizedDimensionList, metricLines []string) []string {
+	summary := metric.Summary()
+
+	for i := 0; i < summary.DataPoints().Len(); i++ {
+		dp := summary.DataPoints().At(i)
+		line, err := serializeSummaryPoint(
+			metric.Name(),
+			prefix,
+			makeCombinedDimensions(defaultDimensions, dp.Attributes(), staticDimensions),
+			dp,
+		)
+
+		if err != nil {
+			logger.Warn(
+				"Error serializing histogram data point",
+				zap.String("name", metric.Name()),
+				zap.Error(err),
+			)
+		}
+
+		if line != "" {
+			metricLines = append(metricLines, line)
+		}
+	}
+	return metricLines
+}
+
+func summaryDataPointToSummary(dp pmetric.SummaryDataPoint) (float64, float64, float64) {
+	var min, max float64
+	if quantileValues := dp.QuantileValues(); quantileValues.Len() > 0 {
+		min = quantileValues.At(0).Value()
+		max = quantileValues.At(quantileValues.Len() - 1).Value()
+	}
```

Comment:
If the quantile values list is length 0 this will set min and max to `0.0` which may cause the data point to be rejected by the server. In other places where we estimate min and max we have added a check to make sure `max >= mean` and `min <= mean`

Reply:
Where this `mean` value comes from? The current code I got from [awsemfexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/datapoint.go#L190), I tried to look there, but I didn't find anything like this.

I don't have much metrics aggregation knowledge, can you point some other place doing similar things?

Reply:
the mean is calculated from the sum and count

Reply:
Should I return something different for min and max when quantile values list is length 0, like the `mean` value?

Reply:
I did a loop on QuantileValues to check for out of order values, don't know if it is possible or not.
Also did the mean checking, see if this is correct.

Reply:
>I did a loop on QuantileValues to check for out of order values, don't know if it is possible or not

I also don't know but I think it is reasonable to guard against and not an expensive loop